src/DependencyInjection/DI.Abstractions/src/ServiceDescriptor.cs

```diff
@@ -107,6 +107,24 @@ private ServiceDescriptor(Type serviceType, ServiceLifetime lifetime)
         /// <inheritdoc />
         public Func<IServiceProvider, object> ImplementationFactory { get; }
 
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var lifetime = $"{nameof(ServiceType)}: {ServiceType} {nameof(Lifetime)}: {Lifetime} ";
+
+            if (ImplementationType != null)
+            {
+                return lifetime + $"{nameof(ImplementationType)}: {ImplementationType}";
+            }
+
+            if (ImplementationFactory != null)
+            {
+                return lifetime + $"{nameof(ImplementationFactory)}: {ImplementationFactory}";
```

Comment:
Is `Func<IServiceProvider, object>.ToString()` ever useful/interesting?

Reply:
I changed it to `ImplementationFactory.Method`